plugins/modules/identity/bitwarden.py

```diff
@@ -0,0 +1,371 @@
+        if self.item.exists:
+            # Edit an existing item based on its current dict.
+            base = self.item.dict
+            bw_args = ['edit', 'item', self.item.id]
+
+        else:
+            # Create a new item based on the template dict.
+            if not self.item.name:
+                raise AssertionError('Name not specified')
+            base = Client.run_json(['get', 'template', 'item'])
+            base['name'] = self.item.name
+            base['login'] = {}
+            base['organizationId'] = self.organization.id
+            base['folderId'] = self.folder.id
+
+            ret_val['changed'] = True
+
+            bw_args = ['create', 'item']
+
+        bw_args += self.organization.bw_filter_args
+
+        # Update base `login` key's contents.
+        for key, value in self._login.items():
+            if do_updates and (key not in base['login'] or base['login'][key] != value):
+                base['login'][key] = value
+                ret_val['changed'] = True
+
+        # Update the base notes section.
+        if do_updates and self._notes is not None:
+            if base['notes'] != self._notes:
+                base['notes'] = self._notes
+                ret_val['changed'] = True
+
+        if ret_val['changed']:
+            # Encode it using bw.
+            out, _stderr = Client.run(['encode'], stdin=json.dumps(base))
+            ret_val['ansible_module_results'] = Client.run_json(bw_args, stdin=out)
+        else:
+            # Return cached value, which should be identical, if no change occured.
+            ret_val['ansible_module_results'] = base
+
+        return ret_val
+
+    def _handle_present_folder(self):
+        ret_val = {'changed': False}
+
+        if self.folder.exists:
+            # A folder with this name already exists; return it.
+            ret_val['ansible_module_results'] = self.folder.dict
+
+        else:
+            # 'No Folder' is a reserved name in Bitwarden -- we shouldn't create a folder
+            # with that name.
+            if self.folder.name in (None, '', 'No Folder'):
+                raise AssertionError('Illegal folder name used')
+
+            # Fill in a template for a folder object to fill in.
+            template = Client.run_json(['get', 'template', 'folder'])
+            template['name'] = self.folder.name
+
+            # Encode it using bw.
+            out, _stderr = Client.run(['encode'], stdin=json.dumps(template))
+
+            # Create the folder.
+            bw_args = ['create', 'folder'] + self.organization.bw_filter_args
+            ret_val['ansible_module_results'] = Client.run_json(bw_args, stdin=out)
+            ret_val['changed'] = True
+
+        return ret_val
+
+    def _handle_present(self, update=True):
+        # Use a lookup table to determine handler for desired target.
+        dispatch = defaultdict(
+            _err('Unknown target'),
+            organization=_err('Creating organizations not supported'),
```

Comment:
:+1: I was trying to program defensively in case someone tried to invoke the API directly from another Python module. I will remove this and anything similar that I can find!